Bugfix for Layer By Time Visibility

Polaris5514 opened the pull request:
<!--

### Requirements

* Filling out the template is required. Any pull request that does not include
enough information to be reviewed in a timely manner may be closed at the
maintainers' discretion.
* Follow the check list items defined by https://github.com/constellation-app/constellation/blob/master/CONTRIBUTING.md#pull-requests
* All new code requires unit tests to ensure they work as expected and will
continue to work as new code is added in the future (regression testing).
* Make sure your branch name is prefixed by `feature`, `bugfix`, `hotfix` or `release`
* Have you read Constellation's Code of Conduct? By filing an issue, you are
expected to comply with it, including treating everyone with respect:
https://github.com/constellation-app/constellation/blob/master/CODE_OF_CONDUCT.md

-->

### Description of the Change
Visibility of Duplicate nodes was being set to 0, changed to 1
<!--

We must be able to understand the design of your change from this description.
If we can't get a good idea of what the code will be doing from the description
here, the pull request may be closed at the maintainers' discretion. Keep in
mind that the maintainer reviewing this PR may not be familiar with or have
worked with the code here recently, so please walk us through the concepts.

-->

### Alternate Designs
N/A
<!--

Explain what other alternates were considered and why the proposed version was
selected.

-->

### Why Should This Be In Core?
Bugfix
<!--

Explain why this functionality should be in Constellation Core as opposed to a
different module suite. Note that this question is more applicable when adding
new functionality. If this change is a minor update to an exising file then it
is understood that this change has to be to this module suite and a response
and therefore a response to this question is not required.

-->

### Benefits
All nodes Visible 
<!-- What benefits will be realized by the code change? -->

### Possible Drawbacks
None realised
<!-- What are the possible side-effects or negative impacts of the code change? -->

### Verification Process

<!--

What process did you follow to verify that your change has the desired effects?

- How did you verify that all new functionality works as expected?
- How did you verify that all changed functionality works as expected?
- How did you verify that the change has not introduced any regressions?

Describe the actions you performed (e.g., buttons you clicked, text you typed,
commands you ran, etc.), and describe the results you observed.

-->

### Applicable Issues
#830 
<!-- Link any applicable issues here -->

arcturus2 reviewed (changes requested):
@Polaris5514 thanks for this and what you have proposed will work. However I was also thinking about just not setting the `vxVisibilityAttr` attribute because it does default to `1`. This would mean that we could delete the definition of `vxVisibilityAttr` and any references to it. If you have tried this already and if that wasn't working then let me know and I'll be happy to approve this.

Polaris5514 commented:
@arcturus2 Apologies, I had attempted this yesterday but Node 0 was still invisible. Have now found that Node 0 is set to invisible on creation of the sphere graph, not within the Layer By Time plugin. 
Requested changes have now been properly implemented

arcturus2 commented on CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/time/LayerByTimePlugin.java:
```diff
@@ -441,7 +437,6 @@ public void read(final GraphReadMethods rg, final PluginInteraction interaction,
                 //Tag on the back of this loop and update the visibility of each node to match it's z value
                 if (isTransactionLayers) {
                     float vxZ = wgcopy.getFloatValue(zAttr, vertexId);
-                    wgcopy.setFloatValue(vxVisibilityAttr, vertexId, vxZ / (z + 1));
                 }
```
@Polaris5514 from my reading of the code I think this entire if block could be removed. What do you think?

arcturus2 reviewed (changes requested):
All good @Polaris5514, you have nothing to apologies for. I think I've found a bit more code that could be deleted so let me know when you have a chance to review.

The advantage of removing code is that it potentially can uncover additional redundant code and takes a step towards removing technical debt which is a good thing 😄

Polaris5514 commented:
Thanks @arcturus2, I have removed the suggested if block. Have also looked through and found two more redundant sections of code and removed them as well.

antares1470 commented on CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/time/LayerByTimePlugin.java:
```diff
@@ -356,8 +355,6 @@ public void read(final GraphReadMethods rg, final PluginInteraction interaction,
             final int timeLayerAttr = wgcopy.addAttribute(GraphElementType.TRANSACTION, LayerNameAttributeDescription.ATTRIBUTE_NAME, LAYER_NAME, "time layer", defaultName, null);
             wgcopy.addAttribute(GraphElementType.GRAPH, IntegerAttributeDescription.ATTRIBUTE_NAME, NLAYERS, "The number of layers to layer by time", 1, null);
             final int zAttr = wgcopy.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "z", "z", 0, null);
```
this variable is apparently no longer being used. If you're able to remove it that would be great.
